errors/error.go

```diff
@@ -174,30 +174,30 @@ func (e *Error) propagateContexts() {
 
 	subBasics := subErr.basics()
 	subExtras := subErr.extras()
+	// If there are no contexts to propagate, return early to avoid race conditions in goroutines when trying to
+	// read/write.
+	if subBasics == nil && subExtras == nil {
+		return
+	}
```

Comment:
This fixes the race condition evident in the unit test mentioned below in the PR 👇